validator/api/src/main/java/tech/pegasys/teku/validator/api/GraffitiManager.java

```diff
@@ -0,0 +1,97 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.service.serviceutils.layout.DataDirLayout;
+
+public class GraffitiManager {
+  static final String GRAFFITI_DIR = "graffiti";
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Path graffitiPath;
+
+  public GraffitiManager(final DataDirLayout dataDirLayout) {
+    this.graffitiPath = createManagementDirectory(dataDirLayout);
+  }
+
+  public Optional<String> setGraffiti(final BLSPublicKey publicKey, final String graffiti) {
+    return updateGraffiti(publicKey, graffiti.strip());
+  }
+
+  public Optional<String> deleteGraffiti(final BLSPublicKey publicKey) {
+    return updateGraffiti(publicKey);
+  }
+
+  private Path createManagementDirectory(final DataDirLayout dataDirLayout) {
+    final Path graffitiDirectory = dataDirLayout.getValidatorDataDirectory().resolve(GRAFFITI_DIR);
+    if (!graffitiDirectory.toFile().exists() && !graffitiDirectory.toFile().mkdirs()) {
+      throw new IllegalStateException(
+          "Unable to create " + GRAFFITI_DIR + " directory for graffiti management.");
+    }
+    return graffitiDirectory;
+  }
+
+  private Optional<String> updateGraffiti(final BLSPublicKey publicKey) {
+    return updateGraffiti(publicKey, "");
+  }
+
+  private Optional<String> updateGraffiti(final BLSPublicKey publicKey, final String graffiti) {
+    final int graffitiSize = graffiti.getBytes(StandardCharsets.UTF_8).length;
+    if (graffitiSize > 32) {
+      return Optional.of(
+          String.format(
+              "'%s' converts to %s bytes. Input must be 32 bytes or less.",
+              graffiti, graffitiSize));
+    }
+
+    try {
+      final Path file = graffitiPath.resolve(resolveFileName(publicKey));
+      Files.writeString(file, graffiti);
+    } catch (IOException e) {
+      final String errorMessage =
+          String.format("Unable to update graffiti for validator %s", publicKey);
+      LOG.error(errorMessage, e);
+      return Optional.of(errorMessage);
+    }
+    return Optional.empty();
+  }
+
+  public Optional<Bytes32> getGraffitiFromStorage(final BLSPublicKey publicKey) {
+    final Path filePath = graffitiPath.resolve(resolveFileName(publicKey));
+    try {
+      return Optional.of(GraffitiParser.loadFromFile(filePath)).filter(this::graffitiNotEmpty);
+    } catch (GraffitiLoaderException | IllegalArgumentException e) {
```

Comment:
The public key will be checked using the key manager in the API `handleRequest(...)` methods. This class just manages saving / getting graffiti. A key would not make it to this manager without being valid.

Reply:
In this method the graffiti is being retrieved and the public key is the one that is input into the Validator when it is created. There is no scenario I can see where the public key would not be the same as one inside a validator (i.e. always valid). This method is used in the new graffiti provider, which is initialised with the public key that is also inputted into the constructor of the validator at the same time.

Reply:
Here is the scenario I am thinking:
1. User uses POST method to create a graffiti for validator x;
2. For some reason, file gets corrupted;
3. User calls GET to retrieve graffiti for validator x;
4. User will get the system-default graffiti (even though the POST request was successful).

From the user point-of-view, the POST is broken because it feels like it did not save his graffiti.

What I would expect is to get a 500 response in the API in this case (from the spec: 	
Internal server error. The server encountered an unexpected error indicative of a serious fault in the system, or a bug).

How do you think we should handle this?